text/0017-parameter_type_with_encoding.md

```diff
@@ -0,0 +1,143 @@
+
+The proposed pattern for the 10 new "cast" and 10 new "bytewise" flags is: `MAV_PARAM_TYPE_[BYTEWISE|CAST]_[UINT8|INT8|UINT16|INT16|UINT32|INT32|UINT64|INT64|REAL32|REAL64]`.
+So for example, to send 32-bit integers you could use `MAV_PARAM_TYPE_INT32`, `MAV_PARAM_TYPE_BYTEWISE_INT32`, or `MAV_PARAM_TYPE_CAST_INT32` (by preference you would send using `MAV_PARAM_TYPE_BYTEWISE_INT32` if this was understood by the GCS.
+
+Question? 
+- I am not sure if we need all these?
+  I imagine the goal is to move to using `MAV_PARAM_TYPE_BYTEWISE` for all types.
+  If you were moving from the existing types to one with encoding in it, when would you ever use the cast type?
+
+
+## GCS Support
+
+A flight stack can't know if it is talking to a recent GCS that understands the new flags (either all of them, or just the subset it supports).
+As a result, if a flight stack sends out the new types, an old GCS will not be able to interpret them. 
+This is a problem because it means that a GCS that could understand the old parameters will not understand the new ones. 
+This problem cannot be fixed by protocol bits, and is in particular difficult when multiple GCS are based off a common source that might take some releases to get into sync.
+
+The proposed soluton is that a bitmask extension field be added to `PARAM_REQUEST_READ` and `PARAM_REQUEST_LIST` which flags which of the new data types the requestor understands.
+The flight stack then configures itself to send the BYTEWISE encoded value of the appropriate type if supported, and otherwise a supported fallback type. 
+
+With the current proposed set of items the flag would have to be a 32-bit bitmask - there will be 10 x 3 parameters.
+
+Flags would be `MAV_PARAM_TYPE_SUPPORTED` values like `MAV_PARAM_TYPE_SUPPORTED_INT32`, `MAV_PARAM_TYPE_SUPPORTED_CAST_INT32`, `MAV_PARAM_TYPE_SUPPORTED_BYTEWISE_INT32`
+
+
+Notes:
+- In future this set may be extended with additional flight modes and component specific modes.
+- The "return home" and "safe recovery" modes both serve the same function: getting the vehicle to a safe location in the event of a failsafe or for landing.
+  They are separated because from the user perspective the behaviour is quite different and the GCS needs to be able to present that behaviour differently.
+  - The existance of these as separate modes means that the autopilot must have separate custom modes for these if we don't publish the standard mode independently. 
+
+## Protocol bits
+
+The two protocol bits `MAV_PROTOCOL_CAPABILITY_PARAM_ENCODE_BYTEWISE` and `MAV_PROTOCOL_CAPABILITY_PARAM_ENCODE_C_CAST` will have to be updated to state that they only apply to the type flags that do not include encoding.
+
+Question 
+- what is the discoverability mechanism for parameter protocol?
+  Right now you should set one of the above bits, and a GCS can use that to determine support.
+  But in future if we all moved to using these new flags, how would we do it?
+
+
+## Deprecation Path
+
+TBD. 
```

Comment:
Would like advice on this. As stated below, it isn't clear to me what the end point is.